controllers/reconcilers.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The Scribe authors.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func createOrUpdateJobRsync(ctx context.Context,
+	l logr.Logger,
+	c client.Client,
+	job *batchv1.Job,
+	owner metav1.Object,
+	scheme *runtime.Scheme,
+	labels map[string]string,
+	envVars []corev1.EnvVar,
+	command []string,
+	dataPVCName string,
+	sshSecretName string,
+	paused bool,
+	saName string) (bool, error) {
+	runAsUser := int64(0)
+	containers := []corev1.Container{{
+		Name:    "rsync",
+		Env:     envVars,
+		Command: command,
+		Image:   RsyncContainerImage,
+		SecurityContext: &corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Add: []corev1.Capability{
+					"AUDIT_WRITE",
+					"SYS_CHROOT",
+				},
+			},
+			RunAsUser: &runAsUser,
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: dataVolumeName, MountPath: mountPath},
+			{Name: "keys", MountPath: "/keys"},
+		},
+	}}
+	secretMode := int32(0600)
+	volumes := []corev1.Volume{
+		{Name: dataVolumeName, VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: dataPVCName,
+			}},
+		},
+		{Name: "keys", VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  sshSecretName,
+				DefaultMode: &secretMode,
+			}},
+		},
+	}
+
+	return createOrUpdateJob(ctx, l, c, job, owner,
+		scheme, labels, containers, volumes, paused, saName)
+}
+
+func createOrUpdateJob(ctx context.Context,
+	l logr.Logger,
+	c client.Client,
+	job *batchv1.Job,
+	owner metav1.Object,
+	scheme *runtime.Scheme,
+	labels map[string]string,
+	containers []corev1.Container,
+	volumes []corev1.Volume,
+	paused bool,
+	saName string) (bool, error) {
+	backoffLimit := int32(2)
+
```

Comment:
I remember we were already talking about this on a call, but you _could_ pull out the common parameters into a struct then that would cut down on the passed params. something like below, 
```bash 
type jobOptions struct {
    ctx context.Context
    l logr.Logger
    c client.Client
    job *batchv1.Job
    owner metav1.Object
    scheme *runtime.Scheme
    labels map[string]string
    paused bool
    saName string
}
```     
since these params are also be passed to `createOrUpdateJobRclone`  and `createOrUpdateJobRestic`